### PR TITLE
feat(mail): Add header arguments to send_mail.

### PR DESCRIPTION
**Pull request by FayeDel**
Add optional header argument from kwargs to positional arguments instead. This uses Django's header property such that extra headers can be easily given, i.e. the X-Auto-Response-Suppress header.

This was added in turn to the attached issue, as the feature would reduce unwanted mails in the mailbox.

Resolves issue: #13784

(For some reason I can't assign reviewers nor ping workflow so I'm not sure what to do)

### CODE REVIEW

**Review by 0ex-d — COMMENTED** [review]
Just curious, can't we not just move all this into `kwargs`?

**Comment by DeltaXWizard** [comment]
In terms of functionality, we probably can move most of the arguments to kwargs anyway.

I wasn't sure why that wasn't done in the first place as my impression was that it was a probable design choice. (IDE autocompletion/ease of use perhaps?)

File: src/sentry/utils/email.py [hunk 1]
```diff
@@ -446,17 +446,23 @@ def get_connection(fail_silently=False):
...
-def send_mail(subject, message, from_email, recipient_list, fail_silently=False, **kwargs):
+def send_mail(
+    subject, message, from_email, recipient_list, fail_silently=False, headers=None, **kwargs
+):
     """
     Wrapper that forces sending mail through our connection.
     Uses EmailMessage class which has more options than the simple send_mail
     """
+    if headers is None:
+        headers = {}
```
Comment (markstory) [inline src/sentry/utils/email.py:457]: I'm not sure an empty dictionary will solve the issue you linked. That issue was also aiming to have auto-reply headers added, which doesn't seem to be part of this change.

Comment (DeltaXWizard) [reply src/sentry/utils/email.py]: Forgot that bit while making the commit, my bad.

File: src/sentry/utils/email.py [hunk 2]
```diff
@@ -446,17 +446,27 @@ def get_connection(fail_silently=False):
...
     """
     Wrapper that forces sending mail through our connection.
     Uses EmailMessage class which has more options than the simple send_mail
     """
+    if headers is None:
+        headers = {
+            "X-Auto-Response-Suppress": "All",
+            "Precedence": "bulk",
+            "Auto-Submitted": "auto-generated",
+        }
```
Comment (dashed) [inline src/sentry/utils/email.py:461]: @DeltaXWizard can you add a test for this change?

Comment (DeltaXWizard) [reply src/sentry/utils/email.py:461]: I don't mind, but I'm not really sure how to write a test using sentry's test utilities with mail headers nor where to start.

Comment (dashed) [reply src/sentry/utils/email.py:461]: @DeltaXWizard You can add a unit test here as a starting point: https://github.com/getsentry/sentry/blob/682684dfb1c7cf6090c753dfff592a4a0d3945be/tests/sentry/utils/email/tests.py#L357-L373
